tracetypes/org.eclipse.tracecompass.incubator.dpdk.core/src/org/eclipse/tracecompass/incubator/internal/dpdk/core/lcore/analysis/LogicalCore.java

```diff
@@ -23,85 +24,174 @@
+
+        private final String fLabel;
+
+        /**
+         * Constructor
+         *
+         * @param label
+         *            String identifying the status of a service
+         */
+        ServiceStatus(String label) {
+            this.fLabel = label;
+        }
+
+        /**
+         * Get the label associated with this service status
+         *
+         * @return the label string
+         */
+        public String getLabel() {
+            return fLabel;
+        }
     }
 
     /**
+     * Assign a role and a status to a logical core
+     *
      * @param ssb
      *            State system builder
      * @param newRole
-     *            Role integer value to input in the state system
+     *            Role to input in the state system
      * @param lcoreId
      *            identifier number for the logical core
      * @param timestamp
      *            time to use for state change
      */
-    public static void setRole(ITmfStateSystemBuilder ssb, Integer newRole, Integer lcoreId, long timestamp) {
-        LogicalCoreStatus newStatus;
-        if (newRole == LogicalCoreRole.ROLE_RTE) {
-            newStatus = LogicalCoreStatus.IDLE;
-        } else if (newRole == LogicalCoreRole.ROLE_OFF) {
-            newStatus = LogicalCoreStatus.OFF;
-        } else {
-            Activator.getInstance().logWarning("LogicalCore setRole with unexpected role value " + newRole.toString()); //$NON-NLS-1$
-            newStatus = LogicalCoreStatus.IDLE;
+    public static void setRole(ITmfStateSystemBuilder ssb, LogicalCoreRole newRole, int lcoreId, long timestamp) {
+
+        // Change the status if the newRole is not service
+        if (newRole != LogicalCoreRole.ROLE_SERVICE) {
+            final LogicalCoreStatus newStatus;
+            if (newRole == LogicalCoreRole.ROLE_RTE) {
+                newStatus = LogicalCoreStatus.IDLE;
+            } else if (newRole == LogicalCoreRole.ROLE_OFF
+                    || newRole == LogicalCoreRole.ROLE_NON_EAL) {
+                newStatus = LogicalCoreStatus.UNKNOWN;
+            } else {
+                Activator.getInstance().logWarning("LogicalCore setRole with unexpected role value!"); //$NON-NLS-1$
+                newStatus = LogicalCoreStatus.UNKNOWN;
+            }
+            setStatus(ssb, newStatus, lcoreId, timestamp);
         }
-        setStatus(ssb, newStatus, lcoreId, timestamp);
-        // Update State system
-        int lcoreQuark = ssb.getQuarkAbsoluteAndAdd(LCORES, String.valueOf(lcoreId));
-        int lcoreRoleQuark = ssb.getQuarkRelativeAndAdd(lcoreQuark, LCORE_ROLE);
-        ssb.modifyAttribute(timestamp, newRole, lcoreRoleQuark);
+
+        // Update the the state system
+        final int lcoreQuark = ssb.getQuarkAbsoluteAndAdd(LCORES, String.valueOf(lcoreId));
+        final int lcoreRoleQuark = ssb.getQuarkRelativeAndAdd(lcoreQuark, LCORE_ROLE);
+        ssb.modifyAttribute(timestamp, newRole.getLabel(), lcoreRoleQuark);
```

Comment:
The state system should ideally only store integer values representing state as it will be much more efficient. In the near future, we might replace these strings with ids automatically in the backend to make implementing analyses easier.
For now, DPDK traces do need much scaling so I guess it can stay as is, if in the future, we need to reduce disk footprint then it might need some improvement.